src/devices/Mpu9250/Mpu9250.cs

```diff
@@ -0,0 +1,983 @@
+            PowerOn();
+            //  PWR_MGMT_1 (0x6B) make CYCLE =0, SLEEP = 0  and STANDBY = 0 
+            WriteRegister(Register.PWR_MGMT_1, (byte)ClockSource.Internal20MHz);
+            // PWR_MGMT_2 (0x6C) set DIS_XA, DIS_YA, DIS_ZA = 0 and DIS_XG, DIS_YG, DIS_ZG = 1
+            // Remove the Gyroscope
+            WriteRegister(Register.PWR_MGMT_2, (byte)(DisableModes.DisableGyroscopeX | DisableModes.DisableGyroscopeY | DisableModes.DisableGyroscopeZ));
+            // ACCEL_CONFIG 2 (0x1D) set ACCEL_FCHOICE_B = 0 and A_DLPFCFG[2:0]=1(b001)
+            // Bandwidth for Accelerator to 184Hz
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth0184Hz;
+            // Enable Motion Interrupt
+            //  In INT_ENABLE (0x38), set the whole register to 0x40 to enable motion interrupt only
+            WriteRegister(Register.INT_ENABLE, 0x40);
+            // Enable AccelHardware Intelligence:
+            // In MOT_DETECT_CTRL (0x69), set ACCEL_INTEL_EN = 1 and ACCEL_INTEL_MODE  = 1
+            WriteRegister(Register.MOT_DETECT_CTRL, 0b1100_0000);
+            // Set Motion Threshold:
+            // In WOM_THR (0x1F), set the WOM_Threshold[7:0] to 1~255 LSBs (0~1020mg)
+            WriteRegister(Register.WOM_THR, (byte)magnetometerThreshold);
+            // Set Frequency of Wake-up:
+            // In LP_ACCEL_ODR (0x1E), set Lposc_clksel[3:0] = 0.24Hz ~ 500Hz
+            WriteRegister(Register.LP_ACCEL_ODR, (byte)acceleratorLowPower);
+            // Enable Cycle Mode (AccelLow Power Mode):
+            // In PWR_MGMT_1 (0x6B) make CYCLE =1
+            WriteRegister(Register.PWR_MGMT_1, 0b0010_0000);
+            // Motion Interrupt Configuration Completed
+        }
+
+        /// <summary>
+        /// Initialize the MPU9250
+        /// </summary>
+        /// <param name="i2cDevice">The I2C device</param>
+        /// <param name="autoDispose">Will automatically dispose the I2C device if true</param>
+        public Mpu9250(I2cDevice i2cDevice, bool autoDispose = true)
+        {
+            _i2cDevice = i2cDevice;
+            _autoDispose = autoDispose;
+            Reset();
+            PowerOn();
+            if (!CheckVersion())
+                throw new IOException($"This device does not contain the correct signature 0x71 for a MPU9250");
+            GyroscopeBandwidth = GyroscopeBandwidth.Bandwidth0250Hz;
+            GyroscopeRange = GyroscopeRange.Range0250Dps;
+            AccelerometerBandwidth = AccelerometerBandwidth.Bandwidth1130Hz;
+            AccelerometerRange = AccelerometerRange.Range02G;
+            // Setup I2C for the AK8963
+            WriteRegister(Register.USER_CTRL, (byte)UserControls.I2C_MST_EN);
+            // Speed of 400 kHz
+            WriteRegister(Register.I2C_MST_CTRL, (byte)I2cBussFrequency.Freqency400kHz);
+            _ak8963 = new Ak8963.Ak8963(_i2cDevice, new Ak8963Attached(), false);
+            Thread.Sleep(50);
```

Comment:
What about this wait? Is it documented?

Reply:
It is not documented. I figured out that a delay needs to be added. Most likely a smaller delay will work. I need to run tests to understand which minimum delay is needed.

Reply:
Found the timing in the documentation: http://www.invensense.com/wp-content/uploads/2015/02/PS-MPU-9250A-01-v1.1.pdf, section 4.23.
In short:
- Maximum risen time is 100 ms after VDD
- Gyroscope = 35 ms before reading after VDD 
- Accelerometer = 30 ms before reading after VDD
- Magnetometer = 7.3 ms after VDD
So let me play with those values to find the optimal one.